common/grid_map_utils/src/polygon_iterator.cpp

```diff
@@ -0,0 +1,180 @@
+// Copyright 2022 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "grid_map_utils/polygon_iterator.hpp"
+
+#include "grid_map_core/GridMap.hpp"
+#include "grid_map_core/Polygon.hpp"
+#include "grid_map_core/TypeDefs.hpp"
+
+#include <grid_map_core/GridMapMath.hpp>
+
+#include <algorithm>
+#include <functional>
+#include <utility>
+
+namespace grid_map_utils
+{
+
+std::vector<Edge> PolygonIterator::calculateSortedEdges(const grid_map::Polygon & polygon)
+{
+  std::vector<Edge> edges;
+  edges.reserve(polygon.nVertices() / 2);
```

Comment:
This is precise number to reserve
```suggestion
  edges.reserve(polygon.nVertices());
```

Reply:
Thank you I applied the change: [513b347](https://github.com/autowarefoundation/autoware.universe/pull/943/commits/513b347d009c5bb3ac815ace1f17db44cfc8a5a5)